tests/plotting/test_plot_manager.py

```diff
@@ -59,23 +60,21 @@ class TestEnvironment:
     dir_2: TestDirectory
 
 
-def create_test_environment(*, dir_1_count: int, dir_2_count: int) -> TestEnvironment:
+@pytest.fixture(scope="function")
+def test_environment(tmp_path) -> TestEnvironment:
+    dir_1_count: int = 7
+    dir_2_count: int = 3
```

Comment:
I hadn't done too much with configurable fixtures so I made up an example with a few different types of uses.  It isn't entirely obviously worth the obfuscation vs. just having a function called in the test or a fixture that provides a callable to call from the test.  But, it does move the setup outside of the test which is "proper".  That way setup failures get you test `ERROR`s rather than test `FAIL`s.  Anyways, here it is.

https://replit.com/@altendky/JudiciousProfuseInfo-2
```python
import dataclasses

import pytest


@dataclasses.dataclass(frozen=True)
class Color:
    red: float
    green: float
    blue: float


# A fixture that provides defaults but can be configured via a mark.
@pytest.fixture
def color(request):
    kwargs = {"red": 0, "green": 0, "blue": 0}
    marker = request.node.get_closest_marker("color_mark")
    kwargs.update(marker.kwargs)

    return Color(**kwargs)


# A test using a default color.
@pytest.mark.color_mark
def test_default_color(color):
    assert color == Color(red=0, green=0, blue=0)


# A test using a configured color.
@pytest.mark.color_mark(red=1, green=0.5, blue=0.7)
def test_b(color):
    assert color == Color(red=1, green=0.5, blue=0.7)


# A test parametrized to run three times, each on a different color.
@pytest.mark.parametrize(
    argnames=["name"],
    argvalues=[
        pytest.param("default", marks=pytest.mark.color_mark),
        pytest.param("blue", marks=pytest.mark.color_mark(red=0, green=0, blue=1)),
        pytest.param("green", marks=pytest.mark.color_mark(red=0, green=1, blue=0)),
        pytest.param("red", marks=pytest.mark.color_mark(red=1, green=0, blue=0)),
    ],
    ids=["default", "blue", "green", "red"]
)
def test_c(name, color):
    assert color == {
        "default": Color(red=0, green=0, blue=0),
        "blue": Color(red=0, green=0, blue=1),
        "green": Color(red=0, green=1, blue=0),
        "red": Color(red=1, green=0, blue=0),
    }[name]



# yuck
if __name__ == "__main__":
    import subprocess
    subprocess.run(["pytest", "--capture=no", "--verbose", "main.py"])
```
```
============================= test session starts ==============================
platform linux -- Python 3.8.12, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- /opt/virtualenvs/python3/bin/python3
cachedir: .pytest_cache
rootdir: /home/runner/JudiciousProfuseInfo, configfile: pytest.ini
collected 6 items                                                              

main.py::test_default_color PASSED
main.py::test_b PASSED
main.py::test_c[default] PASSED
main.py::test_c[blue] PASSED
main.py::test_c[green] PASSED
main.py::test_c[red] PASSED

============================== 6 passed in 0.07s ===============================
```